eel/eel-string.c

```diff
@@ -292,6 +292,10 @@ eel_str_replace_substring (const char *string,
     {
         return NULL;
     }
+    else if (replacement == NULL)
+    {
+        return (char *) string;
```

Comment:
This is *not* good: normally this function returns a newly allocated chunk of memory, and here you just return one of the (constant) arguments.
You should instead return a copy of `string`.

However, the check itself seems wrong: if a `NULL` `replacement` is a valid input for this function, it should be considered the same as a zero-length string, and thus remove occurrences of `substring` from `string`, not merely return an unchanged `string`.

Reply:
it was changed with https://github.com/mate-desktop/caja/commit/3ed089797f798ea47c40d6126d3c002efe81015f